src/lint/validators.nim

```diff
@@ -182,3 +182,20 @@ proc subdirsContain*(dir: string, files: openArray[string]): bool =
         let path = subdir / file
         if not fileExists(path):
           result.setFalseAndPrint("Missing file", path)
+
+proc parseJsonFile*(path: string, b: var bool, checkEmpty = false): JsonNode =
+  ## Parses the file at `path` and returns the resultant JsonNode.
+  ##
+  ## Sets `b` to false if unsuccessful.
+  if fileExists(path):
+    let contents = readFile(path)
+    if not checkEmpty or not isEmptyOrWhitespace(contents):
+      try:
+        result = parseJson(newStringStream(contents), path)
+      except JsonParsingError:
+        b.setFalseAndPrint("JSON parsing error", getCurrentExceptionMsg())
+    else:
+      b.setFalseAndPrint("File is empty, but must contain at least the empty " &
+                         "array, `[]`", path)
```

Comment:
The array suffix not necessarily the right message, as some JSON files should be empty objects then. Maybe just make it a bit more general about the file not being allowed to be empty?

Reply:
Yeah, I know - but we don't have any JSON files where `{}` is valid.

I've tried to improve it in the [force-push diff](https://github.com/exercism/configlet/compare/8d7fd895f17ea16ece7a5d97743b6fd2e6dd4671..ff5349b828a5dbf32cbe2493c5ea10d85d3706e5) (a little noisy, sorry) - it allows us to print a special message for empty `links.json` files, while still printing a "file is empty" message for empty `.meta/config.json` files. Good enough for now?